okhttp/src/jvmMain/kotlin/okhttp3/CacheControl.kt

```diff
@@ -153,11 +150,18 @@ actual class CacheControl internal actual constructor(
 
     actual fun immutable() = commonImmutable()
 
-    actual fun maxAge(maxAge: Int, timeUnit: DurationUnit) = commonMaxAge(maxAge, timeUnit)
+    // We are compiling with kotlin 1.6 but need to run with older versions at runtime. For
+    // maximum compat we therefore need to handle both the case where DurationUnit is typealiased
+    // to TimeUnit (as it was pre-1.6) and where it is a distinct wrapper enum (in 1.6). We also
+    // can't use durationUnit.toTimeUnit() because that doesn't exist in the typealiased case.
+    // This function should work in either case.
+    internal fun toJavaTimeUnit(durationUnit: DurationUnit) = TimeUnit.valueOf(durationUnit.name)
```

Comment:
This doesn't look to bad, a lookup in a Hashmap, from a constant name (at least on JVM).

https://github.com/frohoff/jdk8u-dev-jdk/blob/da0da73ab82ed714dc5be94acd2f0d00fbdfe2e9/src/share/classes/java/lang/Class.java#L3339